percolation/Percolation.java

```diff
@@ -60,154 +61,138 @@ private int rowFromI(int i) {
+                uF.union(i, topi);
 
+            }
 
-        int site1i = iFromCoord(site1Row, site1Col);
-        int site2i = iFromCoord(site2Row, site2Col);
+            if (row < n) {
 
-        int site1id = id[site1i];
-        int site2id = id[site2i];
+                bottomi = iFromCoord(row + 1, col);
 
-        int minID;
-        int maxID;
+                if (opened[bottomi]) uF.union(i, bottomi);
 
-        if (site1id
-                < site2id)  // find minimum id for root to set id values of component sites to (because -1 is the minimum and we want it to spread)
-        {
-            minID = site1id;
-            maxID = site2id;
-        }
-        else {
-            minID = site2id;
-            maxID = site1id;
-        }
+            }
+            else {
+                bottomi = numSites - 1;
 
-        for (int i = 0; i < id.length; i++) {
+                uF.union(i, bottomi);
 
-            if (id[i] == maxID) id[i] = minID;
+            }
 
-        }
+            // continue for topi and bottomi
 
 
+            opened[i] = true;
+            numOpenSites++;
+        }
+
     }
 
     public boolean isOpen(int row, int col)  // is site (row, col) open?
     {
         int i = iFromCoord(row, col);
-        return open1[i];
+        return opened[i];
     }
 
     public boolean isFull(int row, int col)  // is site (row, col) full?
     {
-        if (root(row, col) == topsite) {
-            return true;
-        }
-        else {
-            return false;
-        }
+        int i = iFromCoord(row, col);
+        return (uF.root(i) == uF.root(numSites - 2));
+
     }
 
     public int numberOfOpenSites()       // number of open sites
     {
-        int openSites = 0;
-
-        for (int i = 0; i < open1.length; i++) {
-
-            if (open1[i]) openSites++;
-
-        }
-
-        return openSites;
+        return numOpenSites;
     }
 
     public boolean percolates()              // does the system percolate?
     {
-        int x = id.length;
-        boolean p = false;
-        int lastRow = this.n;
-        int firstCol = 1;
-        int lastRowFirstCol = iFromCoord(lastRow, firstCol);
-
-        for (int i = lastRowFirstCol; i < x; i++) {
-            if (id[i] == -1) {
-                p = true;
-                break;
-            }
-        }
+/*        System.out.println(
+                "Root at topsite is " + Integer.toString(uF.root(numSites - 2))
+                        + " and root at bottom site is "
+                        + Integer
+                        .toString(uF.root(numSites - 1)) + "  and p.percolates is "
+                        + isFull(n + 1, 2));*/
+        return isFull(n + 1, 2);
 
-        return p;
     }
 
     public static void main(String[] args) {
 
-        int N = Integer.parseInt(args[0]);
+        int n = StdIn.readInt();
+
+        Percolation p = new Percolation(n);
 
-        Percolation p = new Percolation(N);
+        p.n = n;
```

Comment:
Shouldn't need this here. If it needs to be set it should be set in Percolation's constructor.